// The greeter service definition.
service Greeter {
  // Sends a greeting
  rpc SayHello (HelloRequest) returns (HelloReply) {}
}

// The request message containing the user's name.
message HelloRequest {
  required string name = 1;
}

// The response message containing the greetings
message HelloReply {
  required string message = 1;
}

message LibraryResourceRequest {
  required string name = 1;
}
message LibraryResource {
  required string name = 1;
  required string content = 2;
}

service LibraryService {
  rpc GetLibraryResource(LibraryResourceRequest) returns (LibraryResource) {}
}
